src/esp/sim/Simulator.h

```diff
@@ -326,6 +359,9 @@ class Simulator {
   core::Random random_;
   SimulatorConfiguration config_;
 
+  std::vector<agent::Agent::ptr> agents_;
```

Comment:
This structure and related functions will not be synchronized with python agents.

Reply:
Yes. Plan was to merge the class first and then merge the functionality. This vector is unused in python since it never uses the C++ addAgent API. Plan is to migrate python over to this API.

Reply:
Some of the functionality will either need to stay in python or will need a bridge or some sort into python.  The ability to add actions in python has been quite useful for instance.

Reply:
The public python API should not change but the internals move to C++. I think that could be accomplished with bridging. We should also support the ability to extend habitat via python. This can accomplished by supporting python sub-classing. One example use case is to create your own Sensor class. 

Overall, I think we should be able to support all current python use cases with bindings and python sub-classing support of some of the C++ classes. I'm hoping none of our users (other than extreme power users) will even notice the change.

But I think it will be extremely valuable to have the core of the simulator in one language instead of 3.